cirq-ft/cirq_ft/algos/apply_gate_to_lth_target.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2023 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+from typing import Callable, Sequence
+
+import attr
+import cirq
+from cirq._compat import cached_property
+from cirq_ft import infra
+from cirq_ft.algos import unary_iteration_gate
+
+
+@attr.frozen
+class ApplyGateToLthQubit(unary_iteration_gate.UnaryIterationGate):
```

Comment:
Is there a reason that this is the "Lth" qubit rather than the "Nth" qubit?  Using n for a number seems to be more standard generally unless this is specifically some sort of QRE standard.

Reply:
This is because the original reference using `L`'th qubit to represent the value stored in selection register. (https://arxiv.org/pdf/1805.03662.pdf)

Added a reference section for more details.